### ISSUE TITLE
tf.nn.l2_normalize takes dim instead of axis

### ISSUE THREAD

**Issue by girving** [tensorflow/tensorflow#14191]
At master, `tf.nn.l2_normalize` still takes the axis parameter as `dim`.  Everything else has been standardized on `axis`, so it'd be nice if this one was `axis` too.  Since `dim` can't be changed for backwards compatibility reasons, presumably the right approach is adding an extra `axis` argument and requiring that at most one of them is set.

@aselle: Is that right?

**Comment by asimshankar** [comment]
That would be correct. Contributions welcome :)

**Comment by yongtang** [comment]
@girving PR #12716 changes dim to axis for `tf.nn.softmax` and `tf.nn.log_softmax`  (with backward-compatibility). The PR is pending review. I can update the PR to include changes of `tf.nn.l2_normalize` from `dim` to `axis` as well.

**Comment by yongtang** [comment]
@girving PR #12716 has been updated to change `dim` to `axis` for `tf.nn.l2_normalize` as well. Please take a look.

**Comment by girving** [comment]
@yongtang Thanks, looks perfect.
